### PR TITLE
Fix guava cache equality issue which causes 0 hit in cache.

### PR DESCRIPTION
**Pull request by yueawang**
## What changes were proposed in this pull request?

1. Fix fiber cache equality issue by overriding hashcode & equals functions.
2. Remove all related magic numbers in fiber cache reading part.
3. Add test case to indicate the equality issue.
4. Add cache size statistics to tell the cache memory capacity/usage for future statistics report.

## How was this patch tested?
mvn test passes.

### CODE REVIEW

File: src/main/scala/org/apache/spark/sql/execution/datasources/oap/filecache/FiberCacheManager.scala [hunk 1]
```diff
@@ -61,6 +62,9 @@ object FiberCacheManager extends Logging {
   private val MB: Double = 1024 * 1024
   private val MAX_WEIGHT = (MemoryManager.maxMemory / MB).toInt
 
+  // cached size
+  private var _cacheSize: Long = 0
```
Comment (LuciferYang) [inline src/main/scala/org/apache/spark/sql/execution/datasources/oap/filecache/FiberCacheManager.scala]: is _cacheSize need ensure threadsafe and use AtomicLong

Comment (yueawang) [reply src/main/scala/org/apache/spark/sql/execution/datasources/oap/filecache/FiberCacheManager.scala]: Yes, it should be, I used synchronize in my local debug commit, but AtomicLong looks much better. I will fix it. Thanks!

File: src/main/scala/org/apache/spark/sql/execution/datasources/oap/filecache/FiberCacheManager.scala [hunk 2]
```diff
@@ -156,6 +176,16 @@ case class BTreeFiber(
...
     idx: Int) extends Fiber {
   override def fiber2Data(conf: Configuration): FiberCache = getFiberData()
+
+  override def hashCode(): Int = (file + section + idx).hashCode
+
+  override def equals(obj: Any): Boolean = obj match {
+    case another: BTreeFiber =>
+      another.section == section &&
+        another.idx == idx &&
+        another.file == file
```
Comment (adrian-wang) [inline src/main/scala/org/apache/spark/sql/execution/datasources/oap/filecache/FiberCacheManager.scala]: better use `equals` for string compare

File: src/main/scala/org/apache/spark/sql/execution/datasources/oap/filecache/FiberCacheManager.scala [hunk 3]
```diff
@@ -17,9 +17,10 @@
 
 package org.apache.spark.sql.execution.datasources.oap.filecache
 
-import com.google.common.cache._
 import java.util.concurrent.{Callable, TimeUnit}
 import java.util.concurrent.atomic.AtomicLong
+
+import com.google.common.cache._
 import org.apache.hadoop.conf.Configuration
 import scala.collection.JavaConverters._
```
Comment (adrian-wang) [inline src/main/scala/org/apache/spark/sql/execution/datasources/oap/filecache/FiberCacheManager.scala]: scala imports also

Comment (yueawang) [reply src/main/scala/org/apache/spark/sql/execution/datasources/oap/filecache/FiberCacheManager.scala]: Where should scala imports go?

Comment (adrian-wang) [reply src/main/scala/org/apache/spark/sql/execution/datasources/oap/filecache/FiberCacheManager.scala]: java, scala, third party, spark

Comment (yueawang) [reply src/main/scala/org/apache/spark/sql/execution/datasources/oap/filecache/FiberCacheManager.scala]: Intellij says "There should at least one a single empty line separating groups java and scala"

File: src/main/scala/org/apache/spark/sql/execution/datasources/oap/filecache/FiberCacheManager.scala [hunk 4]
```diff
@@ -61,6 +62,9 @@ object FiberCacheManager extends Logging {
   private val MB: Double = 1024 * 1024
   private val MAX_WEIGHT = (MemoryManager.maxMemory / MB).toInt
 
+  // Total cached size for debug purpose
+  private val _cacheSize: AtomicLong = new AtomicLong(0)
```
Comment (LuciferYang) [inline src/main/scala/org/apache/spark/sql/execution/datasources/oap/filecache/FiberCacheManager.scala]: If most time we call AtomicLong.add api but need‘t return current value and call AtomicLong.get method low frequency, maybe com.google.common.cache.LongAdder more suitable for use, com.google.common.cache.LongAdder is a  mirror of java 8 java.util.concurrent.atomic.LongAdder,  has better performance in this scene.

Comment (yueawang) [reply src/main/scala/org/apache/spark/sql/execution/datasources/oap/filecache/FiberCacheManager.scala]: Good idea! 
But unfortunately com.google.common.cache.LongAdder is a private class, I think we could move to Java8 LongAdder later.

**Comment by adrian-wang** [comment]
Let's merge this, and leave `LongAdder` as an open issue first.

**Pull request merged**